theano/tensor/slinalg.py

```diff
@@ -413,7 +413,12 @@ def perform(self, node, inputs, outputs):
         U = scipy.linalg.inv(V).T
 
         exp_w = numpy.exp(w)
-        X = numpy.subtract.outer(exp_w, exp_w) / numpy.subtract.outer(w, w)
+        exp_diff = numpy.subtract.outer(exp_w, exp_w)
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", RuntimeWarning)
+            X = exp_diff / numpy.subtract.outer(w, w)
+        X[exp_diff == 0] = 1
```

Comment:
I took a closer look, and I'm not actually sure this is correct.
If we have, for instance, `w[0] == -41` and `w[1] == -42`, then `exp_diff[0, 1] == 0`, but `subtract.outer(w, w)[0, 1] != 0`. Should `X[0, 1]` really be 1 then? Or should we do this instead?

```
diff = numpy.subtract.outer(w, w)
with warnings...:
    X = exp_diff / diff
X[diff == 0] = 1
```